packages/core/src/components/graphs/scatter.ts

```diff
@@ -94,7 +93,10 @@ export class Scatter extends Component {
 			.selectAll("circle.dot")
 			.data(
 				scatterData,
-				(datum) => `${datum[groupMapsTo]}-${datum[domainIdentifier]}`
+				(datum) => {
+					const domainIdentifier = this.services.cartesianScales.getDomainIdentifier(datum);
+					return `${datum[groupMapsTo]}-${datum[domainIdentifier]}`;
+				}
```

Comment:
this might make things a bit slow, wondering if there's a better way to do this since if you have 1000 datapoints this'll run 1000 times